sympy/functions/elementary/exponential.py

```diff
@@ -436,11 +436,15 @@ def _eval_nseries(self, x, n, logx):
         if arg0 in [-oo, oo]:
             return self
         t = Dummy("t")
-        exp_series = exp(t)._taylor(t, n)
+        nterms = n + 2
+        cf = Order(arg.as_leading_term(x), x).getn()
+        if cf != 0:
+            nterms = int(nterms / cf)
+        exp_series = exp(t)._taylor(t, n, nterms)
         o = exp_series.getO()
         exp_series = exp_series.removeO()
         r = exp(arg0)*exp_series.subs(t, arg_series - arg0)
-        r += Order(o.expr.subs(t, (arg_series - arg0)), x)
+        r += Order(x**n, x)
```

Comment:
I am not sure about this line because `r += Order(o.expr.subs(t, (arg_series - arg0)), x)` was returning the O(x**42)